src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -60,6 +59,18 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 		}));
 		this._register(this._inlineChatService.onDidChangeProviders(() => this._onEnablementOrModelChanged()));
 		this._register(this._editor.onDidChangeModel(() => this._onEnablementOrModelChanged()));
+		this._register(this._keybindingService.onDidUpdateKeybindings(() => this._updateDecorationHover()));
+		this._updateDecorationHover();
+		this._onEnablementOrModelChanged();
+	}
+
+	private _updateDecorationHover(): void {
+		const keybinding = this._keybindingService.lookupKeybinding('inlineChat.start')?.getLabel() ?? undefined;
+		if (this._inlineChatKeybinding === keybinding) {
+			return;
+		}
+		this._inlineChatKeybinding = keybinding;
+		this._gutterDecoration.glyphMarginHoverMessage = new MarkdownString(keybinding ? localize('runWithKeybinding', 'Start Inline Chat [{0}]', keybinding) : LOCALIZED_START_INLINE_CHAT_STRING);
 		this._onEnablementOrModelChanged();
```

Comment:
Now `_onEnablementOrModelChanged();` is called twice in a row in the constructor.
Maybe take it out of `_updateDecorationHover()` but add it to the listener
```
this._register(this._keybindingService.onDidUpdateKeybindings(() => { 
   this._updateDecorationHover();
   this._onEnablementOrModelChanged();
}));   
```